samples/wifi/promiscuous/src/main.c

```diff
@@ -46,5 +360,9 @@ int main(void)
 		return ret;
 	}
 
+	k_thread_start(receiver_thread_id);
+
+	k_thread_join(receiver_thread_id, K_FOREVER);
```

Comment:
doesn't this make the thread redundant? This will block the MAIN thread anyway, so, we might as well do the processing here or just exit here and let RX thread keep running?